spring-cloud-stream/src/main/java/org/springframework/cloud/stream/binding/StreamListenerAnnotationBeanPostProcessor.java

```diff
@@ -120,22 +121,6 @@ public void setBeanFactory(BeanFactory beanFactory) throws BeansException {
 		}
 	}
 
-	@SuppressWarnings({ "rawtypes", "unchecked" })
-	@Override
-	public void afterPropertiesSet() throws Exception {
-		Map<String, StreamListenerParameterAdapter> parameterAdapterMap = BeanFactoryUtils
-				.beansOfTypeIncludingAncestors(this.applicationContext, StreamListenerParameterAdapter.class);
-		for (StreamListenerParameterAdapter parameterAdapter : parameterAdapterMap.values()) {
-			this.streamListenerParameterAdapters.add(parameterAdapter);
-		}
-		Map<String, StreamListenerResultAdapter> resultAdapterMap = BeanFactoryUtils
-				.beansOfTypeIncludingAncestors(this.applicationContext, StreamListenerResultAdapter.class);
-		this.streamListenerResultAdapters.add(new MessageChannelStreamListenerResultAdapter());
```

Comment:
I think we need to explicitly register `MessageChannelStreamListenerResultAdapter` as a bean in order to have this bean available during `Lazy` initialization. Also, It might be worth to add a test case to check these required beans available as part of the list of adapters during lazy initialization.

Reply:
Yeah, we need to add _MessageChannelStreamListenerResultAdapter_ in configuration and let Spring manage it.

Reply:
I am about to push an update with `MessageChannelStreamListenerResultAdapter` being injected properly, however I am still not sure where it is used. We seem to be missing a test for this result adapter itself.
In any event we can investigate that later. . .